manifests/mailalias.pp

```diff
@@ -0,0 +1,37 @@
+# == Define: types::mailalias
+#
+define types::mailalias (
+  $ensure    = 'present',
+  $provider  = 'aliases',
+  $recipient = 'MANDATORY',
+  $target    = '/etc/aliases',
+) {
+
+  validate_re($ensure, [ '^present$', '^absent$' ],
+    "types::mailalias::${name}::ensure is invalid and does not match the regex.")
+
+  validate_re($provider, [ '^aliases$' ],
+    "types::mailalias::${name}::provider is invalid and does not match the regex.")
+
+  if $recipient == 'MANDATORY' or empty($recipient) {
+    fail("types::mailalias::${name}::recipient is MANDATORY.")
+  }
+  validate_string($recipient)
+
+  validate_absolute_path($target)
+
+  mailalias { $name:
+    ensure    => $ensure,
+    name      => $name,
+    provider  => $provider,
+    recipient => $recipient,
+    target    => $target,
+    notify    => Exec["${name}-generate_new_aliases_db"],
+  }
+
+  exec { "${name}-generate_new_aliases_db":
```

Comment:
That wouldnt work as it would cause a duplicate resources. The way you have it would work in terms of puppet not having dupes though it would run newaliases multiple times.

Suggest creating a class called mailialiases and including that in the define and then the mailalias resource can notify that exec, similar to how it works now.